doc/source/structures/collections/range.rst

```diff
@@ -0,0 +1,91 @@
+.. _range:
+
+Range
+=====
+
+:struct:`Range` is a type that represents a sequence of scalar whole numbers. The sequence can start and finish at any whole number, can be either descending or ascending and can skip numbers.
+
+.. note::
+
+    This is one of the few places in kOS where there is a distinction between
+    decimal (floating point, or fractional) scalar numbers and whole (integer,
+    or round) scalar numbers.  Using a decimal scalar will not throw an error,
+    however it may give unexpected results due to rounding.
+
+There are 3 ways of constructing a :struct:`Range`:
+
+- :code:`RANGE(FROM, TO, STEP)`
+  Will create a sequence of numbers that starts with `FROM` and contains all numbers `N + STEP` such that `N` belongs to the sequence and that are smaller than `TO`. For example
+  :code:`RANGE(3,8,2)` will contain numbers 3, 5 and 7. If `FROM` > `TO` then the sequence will be descending. For example :code:`RANGE(2,-9,3)` will contain numbers 2, -1, -4 and -7.
```

Comment:
I think this is a case where ambiguity would actually be an improvement.  "Start at 1, Stop at 10" could mean either [1..9] or [1..10] depending on interpretation, or "Begin at 1, End at 10" could mean either [1..9] or [1..10] depending on interpretation, but "From 1 to 10" only means [1..10].  It's an unambiguous meaning, but unfortunately its unambiguously the wrong meaning.  An ambiguous meaning that has to be specified in the docs would be an improvement over an unambiguous one that has to be contradicted in the docs.